core/connection.ts

```diff
@@ -333,6 +333,36 @@ export class Connection implements IASTNodeLocationWithBlock {
     this.createShadowBlock(true);
   }
 
+  /**
+   * Reconnects this connection to the input with the given name on the given
+   * block. If there is already a connection connected to that input, that
+   * connection is disconnected.
+   *
+   * @param block The block to connect this connection to.
+   * @param inputName The name of the input to connect this connection to.
+   * @returns True if this connection was able to connect, false otherwise.
+   */
+  reconnect(block: Block, inputName: string): boolean {
+    // No need to reconnect if this connection's block is deleted.
+    if (this.getSourceBlock().isDeadOrDying()) return false;
+
+    const connectionParent = block.getInput(inputName)!.connection;
```

Comment:
Wouldn't `?.` be safe here?  It seems to me that (especially now this is a method on `Connection` rather than just a local helper function) there's no guarantee there will be an input with the given name.

Of course perhaps you _want_ it to throw if the named input does not exist—but in that case I think an explicit check and `throw new SomeKindOfError('useful message')` would be in order.